src/main/java/org/passay/RuleResultDetail.java

```diff
@@ -30,19 +31,52 @@ public RuleResultDetail(final String code, final Map<String, Object> params)
     if (code == null || code.length() == 0) {
       throw new IllegalArgumentException("Code cannot be null or empty.");
     }
-    errorCode = code;
+    errorCodes = new String[] {code};
+    parameters = params == null ? new LinkedHashMap<>() : new LinkedHashMap<>(params);
+  }
+
+
+
+  /**
+   * Creates a new rule result detail.
+   *
+   * @param  params  error details.
+   * @param  codes  One or more error codes. Codes MUST be provided in order of decreasing specificity.
+   */
+  public RuleResultDetail(final Map<String, Object> params, final String... codes)
```

Comment:
I'd probably forgo varargs here just to keep symmetry with the other map. Plus it just feels like the codes are the first order concern and should come first.

Reply:
Fixed.